components/quick_input/quick_input_state.ts

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+interface ITextboxState{
+    value: string;
+    selectionStart: number | null;
+    selectionEnd: number | null;
+}
+
+const defaultTextboxState: ITextboxState = {
+    value: '',
+    selectionEnd: null,
+    selectionStart: null,
+};
+
+export class QuickInputState {
```

Comment:
Looking at it, I think we might be able to get rid of Quick Input all together and all associated state with it. I think it was used for a bug in an old version of Microsoft Edge that we no longer support. Would you be open to replace it with a HTML input instead?

Reply:
Which means the component quick input is the same, but we do not support rendering custom textbox? Just rendering input only?

Reply:
That could be a challenge to remove `QuickInput`, since some of `SuggestionBox` in the webapp render `QuickInput` with customized components such as `AutosizedTextarea`?

Reply:
Good question! I think this is one of the functionalities that QuickInput shouldn't be responsible for tbh as we already handle some of the logic in Textbox.tsx. @hmhealey can you advice if it's safe to remove QuickInput?

Reply:
So if it's possible I would like to:

1, Remove `QuickInput` from `SuggestionBox` (please advise whether it's safe or not).
2, Remove logic in `QuickInput` that allows rendering custom textbox rather than `input`? (Again, please advise whether it's safe to do so).

Question: Do I need to keep `QuickInputState` as a controlled state of `QuickInput` ? (Personally I like that approach better than passing `onChange(targetEvent)` directly. This practice is used by `Draft-js` of Facebook as well!!)

Suggestion: I would like to convert `SuggestionBox` to controlled component using a controlled state class `SuggestionBoxState`, where it controlled `value`, `suggestions`, `terms` of `SuggestionBox`. With this approach I think we can safely reduce the impact scope of `Providers`. Please lemme know about your opinions.

Reply:
Sounds good to me! It should be safe to remove it as quick input was only implemented for a specific bug in an old version in Microsoft edge. If we run into issues while implementing we can reconsider the approach. 

Regarding quickInputState I don't think it will be necessary as we won't need to store any input specific props? IMO the components state can be handled in SuggestionBox. 

I like the suggestion you can probably store the input/textbox specific props there too.

Reply:
Okay, so far I need to render `inputComponent` inside `SuggestionBox` without going through `QuickInput`, right? This is managable.

Reply:
Correct. And you can go ahead and removing the state as well :)